ui/src/actions/queue.js

```diff
@@ -126,18 +110,20 @@ export function addSamplesToQueue(sampleDataList) {
 export function addSampleAndMount(sampleData) {
   return (dispatch) => {
     dispatch(
-      mountSample(sampleData, () => {
-        sendAddQueueItem([sampleData])
-          .then((json) => {
-            dispatch(setQueue(json));
-            dispatch(selectSamplesAction([sampleData.sampleID]));
-          })
-          .catch((error) => {
-            if (error.status >= 400) {
-              dispatch(showErrorPanel(true, 'Server refused to add sample'));
-            }
-          })
-          .finally(() => dispatch(queueLoading(false)));
+      mountSample(sampleData, async () => {
+        dispatch(queueLoading(true));
+
+        try {
+          const json = await sendAddQueueItem([sampleData]);
+          dispatch(setQueue(json));
+          dispatch(selectSamplesAction([sampleData.sampleID]));
+        } catch (error) {
+          if (error.status >= 400) {
+            dispatch(showErrorPanel(true, 'Server refused to add sample'));
+          }
+        }
+
+        dispatch(queueLoading(false));
```

Comment:
Note that we don't need a `finally` block since the `try` block doesn't return and the `catch` block doesn't re-throw.